src/snapred/backend/recipe/algorithm/ReductionAlgorithm.py

```diff
@@ -182,8 +183,16 @@ def PyExec(self):
         # TODO: Refactor so excute only needs to be called once
         self.mantidSnapper.executeQueue()
 
-        dMin = {pgp.groupID: pgp.dResolution.minimum for pgp in reductionIngredients.pixelGroupingParameters}
-        dMax = {pgp.groupID: pgp.dResolution.maximum for pgp in reductionIngredients.pixelGroupingParameters}
+        # Warning: <pixel grouping parameters>.dResolution will be None for fully-masked groups
+        #   "NaN" is interpreted by 'Mantid::RebinRagged' to indicate an unspecified value
+        dMin = {
+            pgp.groupID: pgp.dResolution.minimum if pgp.dResolution else np.nan
+            for pgp in reductionIngredients.pixelGroupingParameters
+        }
+        dMax = {
+            pgp.groupID: pgp.dResolution.maximum if pgp.dResolution else np.nan
+            for pgp in reductionIngredients.pixelGroupingParameters
+        }
```

Comment:
Is this related to these [similar lines](https://github.com/neutrons/SNAPRed/pull/256/files#diff-a0c5b6461a255c223c08e37521c6b7f014236def75719ef186c4e90fa46e5a21R52-R59)?

if possible should we have some dmin getter method that handles these cases and centralizes this domain knowledge?

Reply:
If we could at least mark these with todo's for when we do figure it out we dont miss them.  Code is best when you dont have to think about these things!

Reply:
I think PR #263 treats both code blocks in the desired way.

Reply:
I think the main difference may be the nan handling

Reply:
I think both this and the other can be moved into the `dMin()`, `dMax()`, and `dBin()` methods in `PixelGroup`